compiler/qsc_ast/src/ast.rs

```diff
@@ -1431,6 +1435,28 @@ impl Default for PathKind {
     }
 }
 
+impl PathKind {
+    /// Returns the segments of the path, whether parsed successfully or not.
+    #[must_use]
+    pub fn segments(&self) -> &[Ident] {
+        match self {
+            PathKind::Ok(path) => path.segments.as_deref().unwrap_or_default(),
+            PathKind::Err(Some(incomplete_path)) => &incomplete_path.segments,
+            PathKind::Err(None) => &[],
+        }
+    }
+
+    /// Returns the name of the path, whether parsed successfully or not.
+    #[must_use]
+    pub fn name(&self) -> Option<&Ident> {
+        match self {
+            PathKind::Ok(path) => Some(&path.name),
+            PathKind::Err(Some(incomplete_path)) => incomplete_path.segments.last(),
+            PathKind::Err(None) => None,
+        }
+    }
+}
+
```

Comment:
I had reservations about adding these methods back when I first defined `PathKind` because I couldn't think of a good reason you'd want to access the segments or name part of the path, without knowing if it's an `Err` or `Ok` first. Besides, there's always the `Idents` impl if you want to straightforwardly iterate over the contained `Ident`s.

On a more practical note, these are unused anyway, so maybe just delete?

Reply:
That makes sense, I've gone ahead and removed them. I didn't need them ultimately, but I thought they might be useful...but dead code is dead code!

Reply:
That makes sense, I've gone ahead and removed them. I didn't need them ultimately, but I thought they might be useful...but dead code is dead code!